Src/Plugins/Security/SEC101_020.DropboxAccessTokenValidator.cs

```diff
@@ -0,0 +1,133 @@
+                                           ref string fingerprint,
+                                           ref string message)
+        {
+            return IsValidStatic(Instance,
+                                 ref matchedPattern,
+                                 ref groups,
+                                 ref failureLevel,
+                                 ref fingerprint,
+                                 ref message);
+        }
+
+        public static string IsValidDynamic(ref string fingerprint, ref string message, ref Dictionary<string, string> options)
+        {
+            return IsValidDynamic(Instance,
+                                  ref fingerprint,
+                                  ref message,
+                                  ref options);
+        }
+
+        protected override string IsValidStaticHelper(ref string matchedPattern,
+                                                      ref Dictionary<string, string> groups,
+                                                      ref string failureLevel,
+                                                      ref string fingerprintText,
+                                                      ref string message)
+        {
+            if (!groups.TryGetNonEmptyValue("refine", out string key))
+            {
+                return nameof(ValidationState.NoMatch);
+            }
+
+            if (!ContainsDigitAndChar(key))
+            {
+                return nameof(ValidationState.NoMatch);
+            }
+
+            fingerprintText = new Fingerprint()
+            {
+                Key = key,
+                Platform = nameof(AssetPlatform.Dropbox),
+            }.ToString();
+
+            return nameof(ValidationState.Unknown);
+        }
+
+        protected override string IsValidDynamicHelper(ref string fingerprintText, ref string message, ref Dictionary<string, string> options)
+        {
+            var fingerprint = new Fingerprint(fingerprintText);
+            string key = fingerprint.Key;
+            using HttpClient httpClient = CreateHttpClient();
+            httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", key);
+
+            try
+            {
+                using HttpResponseMessage response = httpClient
+                    .PostAsync("https://api.dropboxapi.com/2/file_requests/count", null)
+                    .GetAwaiter()
+                    .GetResult();
+
+                switch (response.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                    {
+                        return nameof(ValidationState.AuthorizedError);
+                    }
+
+                    case HttpStatusCode.BadRequest:
+                    {
+                        string body = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+
+                        // App deleted.
+                        if (body.Contains("This app is currently disabled"))
```

Comment:
>Contains [](start = 33, length = 8)

Do we have a more comprehensive string we could encode here? Could this be a StartsWith?

Reply:
updated to ends with. the first piece of the string shows an error message that can be used in other scenarios

---
In reply to: [598008133](https://github.com/microsoft/sarif-pattern-matcher/pull/315#discussion_r598008133) [](ancestors = 598008133)